### PR TITLE
AV-70846 Import sample buckets

### PR DESCRIPTION
**Pull request by Lagher0**
Reverts couchbasecloud/terraform-provider-couchbase-capella#169

### CODE REVIEW

File: internal/schema/sample_bucket_test.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
+package schema
+
+import (
+	"testing"
+
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/errors"
+
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+
+	"github.com/stretchr/testify/assert"
```
[the remark this answers is not in the document]
Comment (Lagher0) [reply internal/schema/sample_bucket_test.go:10]: Could you please explain what would the correct formatting? 

We run our linters with the `golangci-lint run`, right? If so, why doesn't that or the lint check on github pick up the incorrect formatting?

File: internal/resources/sample_bucket.go [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (s *SampleBucket) Update(_ context.Context, _ resource.UpdateRequest, _ *resource.UpdateResponse) {
+	// Couchbase Capella's v4 does not support a PUT endpoint for sample buckets.
+	// SampleBuckets can only be created, read and deleted.
+	// http://cbc-cp-api.s3-website-us-east-1.amazonaws.com/#tag/sampleBucket
```
Comment (Lagher0) [inline internal/resources/sample_bucket.go:206]:
```suggestion
	// http://cbc-cp-api.s3-website-us-east-1.amazonaws.com/#tag/sampleBucket
```

**Review by SaicharanCB — APPROVED** [review]

File: internal/schema/sample_bucket_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (aniket-Kumar-c) [reply internal/schema/sample_bucket_test.go:10]: we have to format import like this

	"context"
	"fmt"
	"net/http"

	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api"
	providerschema "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/schema"

	"github.com/hashicorp/terraform-plugin-framework/datasource"
	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
	"github.com/hashicorp/terraform-plugin-framework/types"

[the remark this answers is not in the document]
Comment (aniket-Kumar-c) [reply internal/schema/sample_bucket_test.go:10]: It is not added by default in any of commands yet

Comment (Lagher0) [inline internal/schema/sample_bucket_test.go:10]:
```suggestion
import (
	"testing"

	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/errors"

	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
	"github.com/stretchr/testify/assert"
```

**Review by aniket-Kumar-c — APPROVED** [review]
LGTM!

**Comment by SaicharanCB** [comment]
+1 from SDET side

**Review by SaicharanCB — APPROVED** [review]

**Comment by SaicharanCB** [comment]
Hi found an edge case here due to this https://couchbasecloud.atlassian.net/browse/AV-75176
We are able to have two or more terraform resources pointing to the same bucket due to this. Though I don't feel it as a blocker

File: examples/getting_started/README.md [hunk 3]
```diff
@@ -1405,6 +1451,30 @@ Terraform will perform the following actions:
...
+          - item_count         = 586 -> null
+          - memory_used_in_mib = 62 -> null
+          - ops_per_second     = 0 -> null
+        } -> null
+      - storage_backend            = "couchstore" -> null
+      - time_to_live_in_seconds    = 0 -> null
+      - type                       = "couchbase" -> null
+    }
+
 Plan: 0 to add, 0 to change, 7 to destroy.
```
Comment (PaulomeeCb) [inline examples/getting_started/README.md:1478]: Don't we need to change this line everywhere in this file, as and when we add the resources?🤔

**Review by PaulomeeCb — COMMENTED** [review]
I think the getting_started folder has the files in such a way that we see the example to create all the resources together through terraform, so the resources are dependent. For example for the bucket, the cluster_id and project_id comes from the couchbase-capella_project.new_project.id and couchbase-capella_cluster.new_cluster.id, instead of a separate variable. Does adding a separate folder also work? CC: @aniket-Kumar-c @nidhi07kumar

**Review by SaicharanCB — DISMISSED** [review]

File: examples/sample_bucket/README.md [hunk 4]
```diff
@@ -0,0 +1,825 @@
...
+
+Sample Output:
+
+```
+Terraform plan 
+╷
+│ Warning: Provider development overrides are in effect
+│ 
+│ The following provider development overrides are set in the CLI configuration:
+│  - couchbasecloud/couchbase-capella in /Users/$USER/workspace/code/Lagher0/terraform-provider-couchbase-capella/bin
```
Comment (PaulomeeCb) [inline examples/sample_bucket/README.md:35]: Make this one general to hide your name -
```suggestion
│  - hashicorp.com/couchbasecloud/capella in /Users/$USER/workspace/terraform-provider-capella
```

File: examples/getting_started/terraform.template.tfvars [hunk 5]
```diff
@@ -94,3 +94,7 @@ app_service = {
     ram = 4
   }
 }
+
+samplebucket = {
```
Comment (PaulomeeCb) [inline examples/getting_started/terraform.template.tfvars:98]: should be 
```suggestion
sample_bucket = {
```

**Review by aniket-Kumar-c — APPROVED** [review]

**Pull request merged**